src/mds/CDentry.cc

```diff
@@ -149,7 +149,7 @@ void CDentry::add_waiter(uint64_t tag, MDSInternalContextBase *c)
 version_t CDentry::pre_dirty(version_t min)
 {
   projected_version = dir->pre_dirty(min);
-  dout(10) << " pre_dirty " << *this << dendl;
+  dout(10) << __func__ << " " << *this << dendl;
```

Comment:
@taodd Are we safe at expanding __func__ all times correctly?

Expanding __func__ at preprocessing time requires the preprocessor to know which function it's processing. The preprocessor generally doesn't know that, because parsing happens after the preprocessor is already done.
Some implementations combine the preprocessing and the parsing, and in those implementations, it would have been possible for __func__ to work the way you'd like it to.